bin/core/src/network/mod.rs

```diff
@@ -0,0 +1,253 @@
+//! # Network Configuration Module
+//! 
+//! This module provides manual network interface configuration for multi-NIC Docker environments.
+//! It allows Komodo Core to specify which network interface should be used as the default route
+//! for internet traffic, which is particularly useful in complex networking setups with multiple
+//! network interfaces.
+//! 
+//! ## Features
+//! - Automatic container environment detection
+//! - Interface validation (existence and UP state)
+//! - Gateway discovery from routing tables or network configuration
+//! - Safe default route modification with privilege checking
+//! - Comprehensive error handling and logging
```

Comment:
Komodo Core uses outbound network for these things
1. Auth providers
2. Git providers
3. Periphery connections

How does setting interface here relate to each of these things? Will it also mean Peripheries are reached via the same interface? I think ideally a user could specify which interface they use for Auth, Git, Periphery individually.

Even if this isn't supported in this PR, I want to make sure this is open for future extension if these features are requested.